inputFiles/compositionalMultiphaseWell/black_oil_wells_saturated_3d_stone2.xml

```diff
@@ -0,0 +1,306 @@
+        distanceFromHead="1.00"/>
+    </InternalWell>
+    <InternalWell
+      name="wellInjector"
+      wellRegionName="wellRegion2"
+      wellControlsName="wellControls2"
+      meshName="mesh"
+      polylineNodeCoords="{ { 195.0, 195.0, 2.0 },
+                            { 195.0, 195.0, 0.0 } }"
+      polylineSegmentConn="{ { 0, 1 } }"
+      radius="0.1"
+      numElementsPerSegment="1">
+      <Perforation
+        name="injectorPerf1"
+        distanceFromHead="1.00"/>
+    </InternalWell>
+  </Mesh>
+
+  <Events 
+      maxTime="3153600">
+    
+    <PeriodicEvent
+      name="outputs"
+      timeFrequency="3153600"
+      targetExactTimestep="1"
+      target="/Outputs/vtkOutput"/>
+    
+    <PeriodicEvent
+      name="timeHistoryOutput"
+      timeFrequency="3153600"
+      targetExactTimestep="1"
+      target="/Outputs/timeHistoryOutput"/>
+
+    <PeriodicEvent
+      name="restarts"
+      timeFrequency="1576800"
+      targetExactTimestep="1"
+      target="/Outputs/restartOutput"/>
+
+    <PeriodicEvent
+      name="solverApplications"
+      maxEventDt="315360"      
+      target="/Solvers/reservoirSystem"/>
+
+    <PeriodicEvent
+      name="statistics"
+      timeFrequency="3153600"
+      target="/Tasks/compositionalMultiphaseFlowStatistics"/>
+    
+    <PeriodicEvent
+      name="timeHistoryCollection"
+      timeFrequency="315360"
+      targetExactTimestep="1"
+      target="/Tasks/wellRateCollection"/>
+
+  </Events>
+
+  <NumericalMethods>
+    <FiniteVolume>
+      <TwoPointFluxApproximation
+        name="fluidTPFA"/>
+    </FiniteVolume>
+  </NumericalMethods>
+
+  <ElementRegions>
+    <CellElementRegion
+      name="region"
+      cellBlocks="{ cb }"
+      materialList="{ fluid, rock, relperm, cappres }"/>
+    <WellElementRegion
+      name="wellRegion1"
+      materialList="{ fluid, relperm }"/>
+    <WellElementRegion
+      name="wellRegion2"
+      materialList="{ fluid, relperm }"/> 
+  </ElementRegions>
+
+  <Constitutive>
+    <BlackOilFluid
+      name="fluid"
+      phaseNames="{ oil, gas, water }"
+      surfaceDensities="{ 800.907131537, 0.856234902739, 1020.3440 }"
+      componentMolarWeight="{ 120e-3, 25e-3, 18e-3 }"
+      tableFiles="{ pvto_bo.txt, pvtg_norv_bo.txt, pvtw_bo.txt }"/>
+          
+    <CompressibleSolidConstantPermeability
+      name="rock"
+      solidModelName="nullSolid"
+      porosityModelName="rockPorosity"
+      permeabilityModelName="rockPerm"/>
+
+    <NullModel
+      name="nullSolid"/>
+
+    <PressurePorosity
+      name="rockPorosity"
+      defaultReferencePorosity="0.2"
+      referencePressure="4.1369e7"
+      compressibility="1.0e-9"/>
+
+    <TableRelativePermeability
+      name="relperm"
+      phaseNames="{ oil, gas, water }"
+      wettingIntermediateRelPermTableNames="{ waterRelPermTable, oilRelPermTableForOW }"
+      nonWettingIntermediateRelPermTableNames="{ gasRelPermTable, oilRelPermTableForOG }"
+      flagInterpolator="1"/>
```

Comment:
can you rename it please into something like `threePhaseModel` ? and make it not 0 and 1 but Baker and Stone2?